<launch>
  <!-- Start the multi-agent navigation stack -->
  <param name="use_sim_time" value="true"/>

  <!-- Launch Gazebo with the specified world -->
  <arg name="robot_type" default="jackal"/>
  <arg name="use_sim_time" default="true" />
  <arg name="headless" default="true" />
  <arg name="verbose" default="true" />
  <arg name="paused" default="false"/>
  <arg name="debug" default="false" />
  <arg name="gui" default="true" />
  <arg name="config1"    default="front_laser"/>
  <arg name="config1_id" default="0"/>
  <arg name="config2"    default="front_laser"/>
  <arg name="config2_id" default="0"/>
  <arg name="config3"    default="front_laser"/>
  <arg name="config3_id" default="0"/>
  <arg name="config4"    default="front_laser"/>
  <arg name="config4_id" default="0"/>
  <arg name="world_name" default="$(find multi_agent_formation)/worlds/square_obstacle_more.world"/>

  <!-- <arg name="world_name" default="$(find jackal_helper)/worlds/jackal_race.world" /> -->

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="headless" value="$(arg headless)"/>
    <arg name="verbose" value="$(arg verbose)"/>
    <arg name="paused" value="$(arg paused)"/>
    <arg name="debug" value="$(arg debug)"/>
    <arg name="gui" value="$(arg gui)"/>
    <arg name="world_name" value="$(arg world_name)" />
  </include>

  <!-- Load jackal robots -->
  <include file="$(find multi_agent_formation)/launch/jackal_base.launch">
      <arg name="ns" value="robot_1" />
      <arg name="config"        value="$(arg config1)"/>
      <arg name="config_id"     value="$(arg config1_id)"/>
      <arg name="use_move_base" value="true"/>
      <arg name="x" value="0.0" />
      <arg name="y" value="0.0" />
      <arg name="yaw" value="0.0" />
      <arg name="joystick" default="false" />
<arg name="use_global_ekf" default="false"/>
  </include>  
  

  <include file="$(find multi_agent_formation)/launch/jackal_base.launch">
      <arg name="ns" value="robot_2" />
      <arg name="config"        value="$(arg config2)"/>
      <arg name="config_id"     value="$(arg config2_id)"/>  
      <arg name="use_move_base" value="false"/>
      <arg name="x" value="2.0" />
      <arg name="y" value="-2.0" />
      <arg name="yaw" value="0.0" />
      <arg name="joystick" default="false" />
  </include>  
  

  <include file="$(find multi_agent_formation)/launch/jackal_base.launch">
      <arg name="ns" value="robot_3" />
      <arg name="config"        value="$(arg config3)"/>
      <arg name="config_id"     value="$(arg config3_id)"/> 
      <arg name="use_move_base" value="false"/>
      <arg name="x" value="0.0" />
      <arg name="y" value="-2.0" />
      <arg name="yaw" value="0.0" />
      <arg name="joystick" default="false" />
  </include>  
  

  <include file="$(find multi_agent_formation)/launch/jackal_base.launch">
      <arg name="ns" value="robot_4" />
      <arg name="config"        value="$(arg config4)"/>
      <arg name="config_id"     value="$(arg config4_id)"/> 
      <arg name="use_move_base" value="false"/>
      <arg name="x" value="2.0" />
      <arg name="y" value="0.0" />
      <arg name="yaw" value="0.0" />
      <arg name="joystick" default="false" />
  </include>  

  <!-- Launch Leader-follower Controllers -->
  <!-- node pkg="multi_agent_formation" name="leader_controller_1" type="leader_controller.py" output="screen"/ -->
  <!-- <node pkg="multi_agent_formation" name="follower_controller_2" type="follower_controller.py" output="screen">
      <rosparam>
          follower_id: 2
      </rosparam>
  </node>
  <node pkg="multi_agent_formation" name="follower_controller_3" type="follower_controller.py" output="screen">
      <rosparam>
          follower_id: 3
      </rosparam>
  </node>
  <node pkg="multi_agent_formation" name="follower_controller_4" type="follower_controller.py" output="screen">
      <rosparam>
          follower_id: 4
      </rosparam>
  </node> -->

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find teb_local_planner)/cfg/rviz_teb_sim.rviz" required="true" />

</launch>
